internal/queryexecutor/range_query_planner.go

```diff
@@ -0,0 +1,117 @@
+package queryexecutor
+
+import (
+	"errors"
+
+	"github.com/IBM-Blockchain/bcdb-server/internal/stateindex"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/constants"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/types"
+)
+
+type rangeQueryPlan struct {
+	startKey *stateindex.IndexEntry
+	endKey   *stateindex.IndexEntry
+}
+
+func createQueryPlan(attribute string, conds *attrConditions) (*rangeQueryPlan, error) {
+	// we assume this function to get only valid conditions
+	//   - eq and no other conditions
+	//   - lt and lte do not appear together
+	//   - gt and gte do not appear together
+
+	if conds.valueType == types.Type_NUMBER {
+		// TODO: support number values
+		return nil, errors.New("numbers are not supported currently")
+	}
+
+	for c := range conds.conditions {
+		if c == constants.NotEqual {
+			// TODO: support not equal
+			return nil, errors.New(constants.NotEqual + " is not supported currently")
+		}
+	}
```

Comment:
Please also comment on that in the previous PR that dissects the query... I was asking my self about this when I read it.

Although I think extending the plan could be easy - check that the NE value is between begin & end, add it ad a list, do the range query, and then filter the results using the NE list, right?

I am OK with doing this in a different PR though.

Reply:
If we ignore the `$neq` first and then apply only after fetching the results, it would be less performant. I am planning to construct multiple range query plans when there is a `$neq` and is more efficient. It will be done as part of https://github.com/IBM-Blockchain/bcdb-server/issues/169